From d14235f9dcd5179561412c6dec951a1a022408f1 Mon Sep 17 00:00:00 2001
From: Rod Whitby <rod@whitby.id.au>
Date: Sun, 27 Jan 2008 10:08:50 +1030
Subject: [PATCH] ixp4xx: Register nslu2 rtc i2c_board_info

Register the i2c board info related to the RTC chip on the nslu2 board
to allow it to be found automatically on boot.

Signed-off-by: Rod Whitby <rod@whitby.id.au>
Signed-off-by: Alessandro Zummo <a.zummo@towertech.it>
---
 arch/arm/mach-ixp4xx/nslu2-setup.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-ixp4xx/nslu2-setup.c b/arch/arm/mach-ixp4xx/nslu2-setup.c
index b10e094..a874a26 100644
--- a/arch/arm/mach-ixp4xx/nslu2-setup.c
+++ b/arch/arm/mach-ixp4xx/nslu2-setup.c
@@ -18,6 +18,7 @@
 #include <linux/serial.h>
 #include <linux/serial_8250.h>
 #include <linux/leds.h>
+#include <linux/i2c.h>
 
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
@@ -47,6 +48,12 @@ static struct ixp4xx_i2c_pins nslu2_i2c_gpio_pins = {
 	.scl_pin		= NSLU2_SCL_PIN,
 };
 
+static struct i2c_board_info __initdata nslu2_i2c_board_info [] = {
+	{
+		I2C_BOARD_INFO("rtc-x1205", 0x6f),
+	},
+};
+
 #ifdef CONFIG_LEDS_IXP4XX
 static struct resource nslu2_led_resources[] = {
 	{
@@ -203,6 +210,9 @@ static void __init nslu2_init(void)
 
 	pm_power_off = nslu2_power_off;
 
+	i2c_register_board_info(0, nslu2_i2c_board_info,
+				ARRAY_SIZE(nslu2_i2c_board_info));
+
 	/*
 	 * This is only useful on a modified machine, but it is valuable
 	 * to have it first in order to see debug messages, and so that
-- 
1.5.2.5

